dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -6620,26 +6473,31 @@ int processWhere(SELECT_LEX &select_lex,
     }
 #endif
 
-    uint32_t failed = buildOuterJoin(gwi, select_lex);
+    uint32_t failed = 0;
 
-    if (failed) return failed;
+    // InfiniDB bug5764 requires outer joins to be appended to the
+    // end of the filter list. This causes outer join filters to
+    // have a higher join id than inner join filters.
+    // TODO MCOL-4680 Figure out why this is the case, and possibly
+    // eliminate this requirement.
+    std::stack<execplan::ParseTree*> outerJoinStack;
 
-    // @bug5764. build outer join for view, make sure outerjoin filter is appended
-    // to the end of the filter list.
-    for (uint i = 0; i < gwi.viewList.size(); i++)
-    {
-        failed = gwi.viewList[i]->processOuterJoin(gwi);
+    if ((failed = buildJoin(gwi, select_lex.top_join_list, outerJoinStack)))
+        return failed;
 
-        if (failed)
-            break;
+    if (gwi.subQuery)
```

Comment:
Why do you decide to put if condition here? I don't see that we specifically process views in buildJoin() so the question where do we handle views w/o subquery naturally arises.

Reply:
Views without subqueries are handled by buildJoin(). If I don't put this if condition here, we will end up creating duplicate filters. For views inside subqueries, however, server does not flatten the join expressions of the views to the outer select and we need access to the subquery view's SELECT_LEX::join_list which is what we achieve with View::processJoin().